madprops/serializers.py

```diff
@@ -61,95 +89,123 @@ class Property(model):
+        # Handle JSON fields (value encoded as JSON).
+        if data['name'] in self.opts.json_props:
+            data['value'] = json.dumps(data['value'])
 
-    def __init__(self, **kwargs):
-        super(NestedPropertySerializer, self).__init__(**kwargs)
-        self.many = True
+        data = self._add_parent_obj_field(data)
+        return super(PropertySerializer, self).to_internal_value(data)
 
-    def _from_native_hook(self, data):
+    def _to_extended_dict(self, data):
+        """ Convert dictionary of properties:
+        {<prop_name>: <prop_value>} ->
+            {'name': <prop_name>, 'value': <prop_value>}
+        """
+        prop_name, prop_value = data.items()[0]
+        return {'name': prop_name, 'value': prop_value}
+
+    def _add_parent_obj_field(self, data):
+        # Property requires ID of parent object (properties owner, e.g. User
+        # for user's properties case). Take it from context, which is passed
+        # from view.
+        parent_obj_field = self.opts.parent_obj_field
+        parent_id_field = parent_obj_field + '_id'
+        data[parent_obj_field] = self.context.get(
+            'parent_id') or self.context['view'].kwargs.get(
+                parent_id_field)
         return data
+
+
+class PropertiesOwnerSerializer(ModelSerializer):
+    """Base class for "parent" serializers.
+
+    If you want to use PropertySerializer with a parent model (e.g. as a
+    nested serializer for User serializer), and have this field writable  -
+    you have to inherit you parent serializer from this class, not
+    ModelSerializer.
+    """
+    def create(self, validated_data):
+        validated_data_minus_properties = dict(validated_data)
+        properties_field = None
+        for (field, serializer) in validated_data:
+            if isinstance(serializer.child, PropertySerializer):
+                del validated_data_minus_properties[field]
+                properties_field = field
+
+        instance = super(PropertiesOwnerSerializer, self).create(
+            validated_data_minus_properties)
+        self._save_properties(instance, properties_field)
+
+    def update(self, instance, validated_data):
+        for attr, value in validated_data.items():
+            if isinstance(self.fields[attr], ListToDictSerializer):
+                self._save_properties(instance, attr)
+                continue
+            setattr(instance, attr, value)
+        instance.save()
+
+        return instance
+
+    def _save_properties(self, instance, field_name):
+        data_dict = self._data_list_to_dict(self.validated_data[field_name])
+        # Get correct serializer class.
+        properties_serializer_class = self.fields[field_name].child.__class__
+        # We can't pass parent ID using View callback like we do for edits.
+        # So we pass parent_id directly.
+        serializer = properties_serializer_class(
+            data=data_dict, many=True, context={'parent_id': instance.pk})
```

Comment:
We could probably extend existing `ListToDictSerializer.validated_data` with parent_id and just call save(), instead of creating new serializer and performing the same validation 2nd time.